tests/integration/tests/test_version_upgrades.py

```diff
@@ -26,10 +26,7 @@ def test_version_upgrades(instances: List[harness.Instance]):
                 "'recent' requires the number of releases as second argument and the flavour as third argument"
             )
         _, num_channels, flavour = channels
-        arch = cp.exec(
-            ["dpkg", "--print-architecture"], text=True, capture_output=True
-        ).stdout.strip()
-        channels = snap.get_latest_channels(int(num_channels), flavour, arch)
+        channels = snap.get_latest_channels(int(num_channels), flavour, cp.arch)
```

Comment:
move determining the arch to the Instance class

Reply:
way better - thanks!